lighthouse-core/audits/third-party-summary.js

```diff
@@ -78,8 +79,11 @@ class ThirdPartySummary extends Audit {
     /** @type {Map<ThirdPartyEntity, {mainThreadTime: number, transferSize: number, blockingTime: number}>} */
     const entities = new Map();
     const defaultEntityStat = {mainThreadTime: 0, blockingTime: 0, transferSize: 0};
+    const mainResource = networkRecords[0];
 
     for (const request of networkRecords) {
+      if (URL.rootDomainsMatch(request.url, mainResource.url)) continue;
+
       const entity = ThirdPartySummary.getEntitySafe(request.url);
```

Comment:
seems like we might rather check if the domain we're auditing is a domain owned by the entity and skip the entire entity?

example: you audit `www.facebook.com` so we ignore those but we still would be surfacing assets from `fbcdn.com` as "third-party"